### ISSUE TITLE
Unable to open certain nedesign .sv6 files (POSIX?)

### ISSUE THREAD

**Issue by magnusg** [OpenRCT2/OpenRCT2#7262]
macOS
10.13.3
0.1.2 build 42490b6

Cannot open old .sv6 files from nedesigns.com or from my old Windows machine.
Saves do open properly on the same installation of RCT2 on my old Windows machine (non-OpenRCT2). Saves from OpenRCT2 can be opened without issues.

Not tested: .sv6 files without custom scenery objects

Help greatly appreciated, as I would really like to pick up the game again on my Mac. If you need further information or if I can be of any help, please let me know.

- [unknown] Reproducible in RCT2 (vanilla)? Do not know, never heard of vanilla as I have not played the game for many years
- [no] Specific to multiplayer? 

**Steps to reproduce:**
1. Start OpenRCT2
2. Load Savegame 
-> returns to start screen of RCT without error code. In version 0.1.11 (4601265) the following error is displayed: Unable to load file ... Game save failed!

**Comment by magnusg** [comment]
Just did some more research on the behaviour, which applies to the same OpenRCT version and build as described above.

Installed OpenRCT2 on a Windows 10 machine and saved the park as a new .sv6 file through OpenRCT2. Then copied the file to my Mac and I can open this new file generated with OpenRCT2 without any problems. 

Hope this information helps finding and resolving the issue.

**Comment by AaronVanGeffen** [comment]
Saving and loading seems to be working fine for me on macOS. Could you attach one of the malfunctioning saves here? (You'll have to zip them for GitHub.)

> [unknown] Reproducible in RCT2 (vanilla)? Do not know, never heard of vanilla as I have not played the game for many years

'Vanilla' is how we refer to the original RCT2 game, with or without expansion packs.

**Comment by magnusg** [comment]
Here are two files not working for me, only on macOS:
[H2H6 Starter Bench.SV6.zip](https://github.com/OpenRCT2/OpenRCT2/files/1800481/H2H6.Starter.Bench.SV6.zip)
[[H2H6]Flying Germans - Park Edda.SV6.zip](https://github.com/OpenRCT2/OpenRCT2/files/1800482/H2H6.Flying.Germans.-.Park.Edda.SV6.zip)

Here is the .sv6 that I saved out of OpenRCT2, which works fine. The file size is about 2MB larger for this file. I wonder if the export of custom objects is part of the problem, but as I have all the objects ObjData folder it should not cause this issue.
[[H2H6]Flying Germans - Park Edda_exported.sv6.zip](https://github.com/OpenRCT2/OpenRCT2/files/1800488/H2H6.Flying.Germans.-.Park.Edda_exported.sv6.zip)

**Comment by AaronVanGeffen** [comment]
Hmm, I'm getting an "Unable to detect file type." error in stdout when I try to load the first two saves. The third save works fine.

I'll investigate what's going on.

**Comment by magnusg** [comment]
The first file has been saved with the original RCT2 game some years ago on a WinXP or Win7 computer. I now went through most of my old saves and none of them works on the MacOS OpenRCT2 installation. 
For the first file I am the original creator (most credit goes to robbie92 though), but I think posix made the final saved version of it, before uploading to nedesigns. 
I also did some testing on linux which gives me the same problems I encountered on MacOS.

Another strange behaviour I found related to this is the following:
1. Open any park from OpenRCT2
2. Open another .sv6 file from the menu which does not work (see previous uploads) -> Park does not open
3. Save As dialog refers to the file I tried to open in step 2.

**Comment by AaronVanGeffen** [comment]
Neither of the files works on Windows for me, either. Both the 32 and 64-bits give me the same error.

**Comment by magnusg** [comment]
Very strange. Works fine for me on Windows 10. I have the files sitting on the original RCT2 installation and not in the OpenRCT2 folder. 
I tried opening the .sv6 files from the installation path saved games folder of RCT2 (not OpenRCT2) and it works fine on macOS and linux now. The problem appears to be caused by the copy process from the RCT2 installation path to the OpenRCT2 save folder.

Hope this helps fixing the issue.

**Comment by IntelOrca** [comment]
H2H6 Starter Bench.SV6 is only 6 KB and starts with a invalid chunk. I would be surprised if this file loaded in anything.

**Comment by magnusg** [comment]
Agreed.
I copied the complete RCT2 folder from my old Windows machine to my MacBook and used it as the RCT2 install path. 
The default path from OpenRCT2 to open savedgames is the saves folder from the OpenRCT2 installation. The savedgames from the RCT2 folder are copied by the OpenRCT2 installer. This copy process appears to be corrupted, as I can open all the .sv6 files from the RCT2 folder, but the files from the OpenRCT2 saves folder are only 6KB and therefore not working.

In other words, I found how to work around the issue. However, there appears to be an issue with the copy process from the installer.

Thanks for all your input. It looks like I can finally enjoy the game on my MacBook. Even the old  Thrillmatic2 park works again and it is a pleasure viewing the old parks after so many years. :)

**Comment by AaronVanGeffen** [comment]
Glad to hear this resolves the matter for you. If you can reproduce the issue with the installer, that is certainly something we should have a look at. If you have the time to investigate, please open another ticket for it.

**Comment by Gymnasiast** [comment]
Shouldn't we check if copying the files over is broken on POSIX?

**Comment by janisozaur** [comment]
Also: trying to open those files results in no park being loaded at all. I.e. black screen with just the FPS counter, if enabled.

**Comment by AaronVanGeffen** [comment]
> Shouldn't we check if copying the files over is broken on POSIX?

Allegedly this was caused by the installer, though — which is Windows-only, right?

> Also: trying to open those files results in no park being loaded at all. I.e. black screen with just the FPS counter, if enabled.

The files are damaged. Only loading straight from the command line does it result in a black screen; loading from the game itself just resets the title screen.
